### ISSUE TITLE
EnumTypeWrapper objects should be usable according to the enum.Enum interface (or maybe the enum.IntEnum interface)

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [protocolbuffers/protobuf#5510]
**To what language does this apply?** Python.

**Describe the problem you are trying to solve.**
The Python language (since 3.4) has an `enum` module, but the generated Python classes in `_pb2` files don't behave according to the interfaces of the classes defined in the `enum` module.

**Describe the solution you'd like**
I think `EnumTypeWrapper` objects should support being used, as much as possible, as though they were subclasses of `enum.Enum`.

**Describe alternatives you've considered**
`enum.IntEnum`, though disfavored in new code, exists to allow `int`-based ad hoc "enums" from before Python 3.4 to be upgraded to be usable according to the interface of `enum.Enum`. It may have a role to play.

**Additional context**
This would be really great for type-checking; it would allow authors to describe parameters and return values as of type `my_module_pb2.MyEnum` rather than as of type `int`.

**Comment by thisch** [comment]
See https://github.com/protocolbuffers/protobuf/pull/5234, where I asked whether the EnumTypeWrapper classes can be replaced by enum from the stdlib.

**Comment by nathanielmanistaatgoogle** [comment]
> We triage inactive PRs and issues in order to make it easier to find active work. If this issue should remain active or becomes active again, please add a comment.
>
> This issue is labeled inactive because the last activity was over 90 days ago.

Yes, this issue still tracks a fantastic opportunity for improvement in Protocol Buffers Python.

**Comment by AlexeyGy** [comment]
I would like to take this on if it's still unassigned.

**Comment by cvrebert** [comment]
Still a valid, not-yet-implemented improvement suggestion, Mr. Robot.

**Comment by arthur-tacca** [comment]
still active

**Comment by cvrebert** [comment]
Still desirable.

**Comment by jguamie** [comment]
This is not planned at this time. However, we will consider this in a next Protobuf planning cycle.

**Comment by esrauchg** [comment]
Looking closer, I believe this issue is actually closed as fixed.

In the years since this bug was created, we added enum info to the .pyi for typed enums, which you can use `--pyi_out` flag to get those type bindings.

The types do _not_ use enum.Enum intentionally; the reason is that Protobuf Enums in Proto3 and by-default going forward with Editions are open enums: any int value is valid, just some of them happen to have names. 

he enum.Enum behavior unfortunately will convey closed enum semantics (where you expect only named values to ever exist), which was the Proto2 behavior and is something you can opt-into with Editions, but has been found to have very undesirable properties for enums in a wire format (as seen at "https://protobuf.dev/programming-guides/enum/").

Since most OSS usage is on Proto3 and open enums are also intended to be the strongly preferred default going forward, we don't intend to go far in better typing only on closed enums.
